idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+					npix=histogram(hd[pos[*,0],pos[*,1]],binsize=1)
+					magpol=indgen(max(hd[pos[*,0],pos[*,1]])-min(hd[pos[*,0],pos[*,1]]))+min(hd[pos[*,0],pos[*,1]])
+
+					wh=where(npix eq 0)
+					if wh[0] ne -1 then npix[where(npix eq 0)]=1
+
+;=====magnetic cut offs dependant on area=========
+					if abs((total(npix[where(magpol gt 0)])-total(npix[where(magpol lt 0)]))/sqrt(total(npix))) ge 10 or arcar gt 9000 then begin 
+						if abs(mean(hd[pos[*,0],pos[*,1]])) gt garr[cent[0],cent[1]] or arcar gt 40000 then begin
+
+;====create an accurate center point=======
+							ypos=total((where(iarr eq ident)/s[1])*abs(lat(where(iarr eq ident))))/total(abs(lat(where(iarr eq ident))))
+							xpos=total((((where(iarr eq ident)/float(s[1]))-(where(iarr eq ident)/s[1]))*s[1])*abs(lon(where(iarr eq ident))))/total(abs(lon(where(iarr eq ident))))
+							arccent0=coord[0,xpos,ypos]
+							arccent1=coord[1,xpos,ypos]
+
+;======calculate average angle coronal hole is subjected to======
+							dist=sqrt((arccent0)^(2)+(arccent1)^(2))
+							ang=2*asin(SQRT(dist/(2*rs)))
+
+;=====calculate area of CH with minimal projection effects======
+							trupixar=abs(area/cos(ang))
+							truarcar=trupixar*dattoarc*dattoarc
+							trummar=truarcar*(6.96e+08/rs)*(6.96e+08/rs)
+
+;====find CH extent in lattitude and longitude========
+							maxxlat=lat[max(xy[0,info(i).offset+chpts]),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]]
+							maxxlon=lon[max(xy[0,info(i).offset+chpts]),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq max(xy[0,info(i).offset+chpts])))]]
+							maxylat=lat[xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq max(xy[1,info(i).offset+chpts])))],max(xy[1,info(i).offset+chpts])]
+							maxylon=lon[xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq max(xy[1,info(i).offset+chpts])))],max(xy[1,info(i).offset+chpts])]
+							minxlat=lat[min(xy[0,info(i).offset+chpts]),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq min(xy[0,info(i).offset+chpts])))]]
+							minxlon=lon[min(xy[0,info(i).offset+chpts]),xy[1,info(i).offset+min(where((xy[0,info(i).offset+chpts]) eq min(xy[0,info(i).offset+chpts])))]]
+							minylat=lat[xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq min(xy[1,info(i).offset+chpts])))],min(xy[1,info(i).offset+chpts])]
+							minylon=lon[xy[0,info(i).offset+min(where((xy[1,info(i).offset+chpts]) eq min(xy[1,info(i).offset+chpts])))],min(xy[1,info(i).offset+chpts])]
+
+;=====CH centroid in lat/lon=======
+							centlat=lat[cent[0],cent[1]]
+							centlon=lon[cent[0],cent[1]]
+
+;====caluclate the mean magnetic field=====
+							mB=mean(hd[pos[*,0],pos[*,1]])
+							if mB ge 0 then smb=4 else smb=5
+							mBpos=total(npix[where(magpol gt 0)]*magpol[where(magpol gt 0)])/total(npix[where(magpol gt 0)])
+							mBneg=total(npix[where(magpol lt 0)]*magpol[where(magpol lt 0)])/total(npix[where(magpol lt 0)])
```

Comment:
⚠️ - more use of `where`. This could produce bad stuff.
You can  in a single call of `where` get also the complementary.
Additionally, with so many `where` in the code this could go very slow. Check why [the histogram function is amazing](http://www.idlcoyote.com/tips/histogram_tutorial.html).